Providers/BotInfoProvider.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+
+namespace Theorem.Providers
+{
+    /// <summary>
+    /// SlackProvider provides all Slack functionality (send/receive/etc)
+    /// </summary>
+    public class BotInfoProvider
```

Comment:
Should MiddlewarePipeline be refactored to utilize this class? And is this provider necessary over just injecting the list of middlewares into classes directly..?